src/AzureClient/Visualization/CloudJobEncoders.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.Azure.Quantum;
+using Microsoft.Jupyter.Core;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Quantum.IQSharp.AzureClient
+{
+    internal static class CloudJobExtensions
+    {
+        internal static Dictionary<string, object> ToDictionary(this CloudJob cloudJob) =>
+            new Dictionary<string, object>()
+            {
+                // TODO: add cloudJob.Uri after https://github.com/microsoft/qsharp-runtime/issues/236 is fixed.
+                { "id", cloudJob.Id },
+                { "name", cloudJob.Details.Name },
+                { "status", cloudJob.Status },
+                { "provider", cloudJob.Details.ProviderId },
+                { "target", cloudJob.Details.Target },
+            };
+
+        internal static Table<CloudJob> ToJupyterTable(this IEnumerable<CloudJob> jobsList) =>
+            new Table<CloudJob>
+            {
+                Columns = new List<(string, Func<CloudJob, string>)>
+                {
+                    // TODO: add cloudJob.Uri after https://github.com/microsoft/qsharp-runtime/issues/236 is fixed.
+                    ("Job ID", cloudJob => cloudJob.Id),
+                    ("Job Name", cloudJob => cloudJob.Details.Name),
+                    ("Job Status", cloudJob => cloudJob.Status),
+                    ("Provider", cloudJob => cloudJob.Details.ProviderId),
+                    ("Target", cloudJob => cloudJob.Details.Target),
```

Comment:
Should we include submission/completion times?

Reply:
Yes, I think that's a great idea.